### ISSUE TITLE
Programmatically start a stack from a catalog?

### ISSUE THREAD

**Issue by mattdodge** [rancher/rancher#8181]
Is it possible to launch a new stack from a catalog entry (public or private catalog) via the API? So far, I've only been able to find the docs on starting a stack generically like so:

```
POST /v2-beta/projects/env/stacks
{
  "dockerCompose": "docker_compose_contents_here",
  "environment": {
    "RANCHER_COMPOSE_QUESTION_1": "RANCHER_COMPOSE_ANSWER_1"
  },
  "externalId": "catalog://my_catalog:my_stack:0",
  "rancherCompose": "rancher_compose_contents_here"
}
```

However, doing this requires that I know the contents of my docker-compose and rancher-compose files. I would rather keep those versioned in the catalog repository and just pass the answers to the stack creation questions to launch a new stack. If I pass null to `dockerCompose` and `rancherCompose` I get the stack but no services appear in it (makes sense).

I think I can use `rancher-compose` CLI to do this but given we'll be doing this from an application I would rather use the API if possible. Am I missing an endpoint somewhere?

**Comment by vincent99** [comment]
Creating a stack using a catalog item just reads the yaml from the catalog and sends it to that endpoint; there is no special communication between the catalog service and cattle, or way to just say "just go deploy this catalog item" by name/id.

**Comment by mattdodge** [comment]
Ah ok, that makes sense I guess. In that case, the question is more "how can I programmatically fetch the contents of docker-compose.yml and rancher-compose.yml" then. 

I'm not able to find any official documentation on it, but through [some old](https://forums.rancher.com/t/rancher-catalog-api/3110) [forum posts](https://forums.rancher.com/t/use-api-to-start-new-container-from-catalog/3084/3) I discovered a "catalog API" which is hosted at `/v1-catalog`. Is this what we should be using to do this? 

No docs, but by clicking around I was able to find the templates I need here:
```
GET /v1-catalog/templates/my_catalog:my_stack:0

Response:
{
  ...
  files: {
    "docker-compose.yml": "---contents---",
    "rancher-compose.yml": "---contents---"
  },
  ...
}
```

I'm all for using APIs that aren't documented or productized yet, but just want to make sure that's the best bet for me. Especially given the `v1` prefix, I'm a little uncertain.
